ecommerce/views.py

```diff
@@ -40,3 +39,114 @@ def update_profile(request):
         user_form = UserForm(instance=request.user)
         profile_form = ProfileForm(instance=request.user.profile)
     return render(request, 'ecommerce/profile.html', {'user_form': user_form,'profile_form': profile_form})
+
+@login_required
+@transaction.atomic
+def cart_add(request, pk):
+    if request.method == 'POST':
+        quantity = request.POST['quantity']
+    else:
+        quantity = 0
+
+    product_get = Product.objects.get(pk=pk)
+    if product_get:
```

Comment:
Product.objects.get(pk=pk) => sẽ raise exception "DoesNotExist" thì check if như này ko được đâu em nha
Em chuyển sang dùng first() để nó return về None nếu ko có nha:
```suggestion
    product_get = Product.objects.filter(pk=pk).first()
    if product_get:
```